### PR TITLE
FreeBSD / jail documentation

### PR DESCRIPTION
**Pull request by FeudalNoodle**
The first draft for FreeBSD / jail installation documentation. I tried to match the visual style, but it still might need some small tweaks.
Some sections could be slightly expanded, but I expect the few edge cases who end up using this guide to at least know a little bit about what they are doing.

### CODE REVIEW

File: tools/Documentation/installing-lanraragi/jail.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+# FreeBSD / Jail
+
+Installing LANraragi on FreeBSD or in a jail is similar to [installing it from source](https://sugoi.gitbook.io/lanraragi/installing-lanraragi/source) with slighly altered dependecies and some extra steps.
+
+## Jail creation
+
+For creating the jail on a regular FreeBSD installation refer to its [documentation](https://docs.freebsd.org/doc/7.3-RELEASE/usr/share/doc/handbook/jails-build.html).
+
+For creating the jail on FreeNas simply navigate to the jails tab in the webui and click on 'add'.
```
Comment (Difegue) [inline tools/Documentation/installing-lanraragi/jail.md:14]:
```suggestion
For creating the jail on FreeNAS, simply navigate to the jails tab in the webui and click on 'add'.
```

File: tools/Documentation/installing-lanraragi/jail.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  If your server is running on FreeBSD (FreeNAS/TrueNAS for example) and you do not want to use virtualisation you can install LANraragi in a jail. This process is rather cli heavy and depending on you and your system error prone so it is not recommended for beginners or casual users.
+---
+
+# FreeBSD / Jail
+
+Installing LANraragi on FreeBSD or in a jail is similar to [installing it from source](https://sugoi.gitbook.io/lanraragi/installing-lanraragi/source) with slighly altered dependecies and some extra steps.
+
+## Jail creation
+
+For creating the jail on a regular FreeBSD installation refer to its [documentation](https://docs.freebsd.org/doc/7.3-RELEASE/usr/share/doc/handbook/jails-build.html).
```
Comment (Difegue) [inline tools/Documentation/installing-lanraragi/jail.md:12]:
```suggestion
For creating the jail on a regular FreeBSD installation, refer to its [documentation](https://docs.freebsd.org/doc/7.3-RELEASE/usr/share/doc/handbook/jails-build.html).
```

File: tools/Documentation/installing-lanraragi/jail.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
+---
+description: >-
+  If your server is running on FreeBSD (FreeNAS/TrueNAS for example) and you do not want to use virtualisation you can install LANraragi in a jail. This process is rather cli heavy and depending on you and your system error prone so it is not recommended for beginners or casual users.
```
Comment (Difegue) [inline tools/Documentation/installing-lanraragi/jail.md:3]:
```suggestion
  If your server is running on FreeBSD (FreeNAS/TrueNAS for example) and you do not want to use virtualization, you can install LANraragi in a jail. This process is rather cli-heavy and error prone (depending on your system), so it is not recommended for beginners or casual users.
```

File: tools/Documentation/installing-lanraragi/jail.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+![Jail Overview](../.gitbook/assets/jails.jpg)
+
+After creating the jail enter it, type ```pkg``` and confirm the next prompt.
+
+![Entering a jail](../.gitbook/assets/shell.jpg)
+
+If you want to install it on the main system itself (which in most cases is not recommended) you can simply skip this step.
+
+## Installing dependecies
```
Comment (Difegue) [inline tools/Documentation/installing-lanraragi/jail.md:24]:
```suggestion
## Installing dependencies
```

File: tools/Documentation/installing-lanraragi/jail.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+pkg install libarchive
+pkg install ImageMagick7
+pkg install libressl
+pkg install npm
+pkg install p5-mojolicious
+pkg install git
+cpan Parallel::Loops
+```
+
+Afterwards you need to add redis_enable="YES" to your rc.conf with your text editor of choice. The preinstalled one is easy editor but you can install nano as well.
```
Comment (Difegue) [inline tools/Documentation/installing-lanraragi/jail.md:44]:
```suggestion
Afterwards, you need to add redis_enable="YES" to your `rc.conf` with your text editor of choice. The preinstalled one is Easy Editor (`ee`), but you can install `nano` as well.
```

**Review by Difegue — CHANGES_REQUESTED** [review]
Got a few grammar suggestions. 👍 

To make this new page visible in the docs, you also need to add a link to it in https://github.com/Difegue/LANraragi/blob/dev/tools/Documentation/SUMMARY.md.  

It'd be nice to add a link under https://github.com/Difegue/LANraragi/blob/dev/tools/Documentation/installing-lanraragi/methods.md#linuxcommunity-community-provided-install-packages as well. 

Thanks again! ❤️

File: tools/Documentation/installing-lanraragi/jail.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+pkg install gnupg
+pkg install p5-App-cpanminus
+pkg install redis
+pkg install libarchive
+pkg install ImageMagick7
+pkg install libressl
+pkg install npm
+pkg install p5-mojolicious
+pkg install git
+cpan Parallel::Loops
```
[the remark this answers is not in the document]
Comment (FeudalNoodle) [reply tools/Documentation/installing-lanraragi/jail.md:41]: For whatever reason the installer script is unable to install any pearl modules on its own. If you do not install it manually npm just exits in the middle of installation due to Parallel::Loops missing.
Might be a separate issue with the installer script. Could be worth investigating to increase compatibility out of the box.

[the remark this answers is not in the document]
Comment (Difegue) [reply tools/Documentation/installing-lanraragi/jail.md:41]: The installer normally does install modules on its own through `cpanm` just fine, does the issue only happen with Parallel::Loops?  
If it installs no modules at all, tbh the entire install is going to end up broken in multiple ways...

[the remark this answers is not in the document]
Comment (FeudalNoodle) [reply tools/Documentation/installing-lanraragi/jail.md:41]: I am not sure. Parallel::Loops is the perl module I must install separately. On my first install I had more issues with dependencies. If there was one missing, which had an appropriate perl module as substitute, it was unable to install that one as well.
Second install on a fresh jail with the aforementioned dependencies runs fine. Except from the one small npm error related to mojolicious on the first server startup.
After migrating my current LANraragi stuff over it seems fully functional up until now.

[the remark this answers is not in the document]
Comment (FeudalNoodle) [reply tools/Documentation/installing-lanraragi/jail.md:41]: A small note on a dependency: On FreeBSD you specifically need ImageMagick7. Don't know if it is the same for Linux in general, but ImageMagick6 misses some libraries that LANraragi needs and causes fails on installation.

File: tools/Documentation/installing-lanraragi/methods.md [hunk 7]
```diff
@@ -50,6 +50,10 @@ Ready-to-install packages provided by voluntary maintainers or by a linux distri
 
 {% page-ref page="community.md" %}
 
+## FreeBSD/Jail
+
+Similar to installing from source with an altered process for FreeBSD compatability.
+
```
Comment (Difegue) [inline tools/Documentation/installing-lanraragi/methods.md:56]: Add a page-ref here so it makes a link in the docs:  

```suggestion
{% page-ref page="jail.md" %}
```

**Review by Difegue — CHANGES_REQUESTED** [review]

File: tools/Documentation/installing-lanraragi/methods.md (hunk 7 above)
Comment (FeudalNoodle) [reply tools/Documentation/installing-lanraragi/methods.md:56]: Whoops, completely forgot that.

File: tools/Documentation/installing-lanraragi/jail.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (FeudalNoodle) [reply tools/Documentation/installing-lanraragi/jail.md:41]: Just thought about having to mention it in the Source Install on Linux section, but I guess most package managers get the lastest version automatically. In pkg you must specify 6 or 7 not specifying doesn't work.

[the remark this answers is not in the document]
Comment (Difegue) [reply tools/Documentation/installing-lanraragi/jail.md:41]: Most distros seem to have moved on to 7 already(even Debian iirc) so I didn't consider it needed, but if you want to add a blerb about it in the source install page that's fine by me. 👍

**Review by Difegue — APPROVED** [review]

**Comment by Difegue** [comment]
Looks good! You should be able to see the changes once merged at https://sugoi.gitbook.io/lanraragi/v/dev/ . 👏

**Pull request merged**